src/presenter/trip-presenter.js

```diff
@@ -5,19 +5,70 @@ import PointsList from '../view/points-list-view.js';
 import {render} from '../render.js';
 
 export default class TripPresenter {
-  tripList = new PointsList();
 
-  init = (tripContainer, pointsModel) => {
-    this.tripContainer = tripContainer;
-    this.pointsModel = pointsModel;
-    this.tripPoints = [...this.pointsModel.getPoints()];
+  #tripContainer = null;
+  #pointsModel = null;
+  #tripPoints = [];
+  #tripList = new PointsList();
 
-    render(new SortView(), this.tripContainer);
-    render(this.tripList, this.tripContainer);
-    render(new PointEditView(this.tripPoints[0]), this.tripList.getElement());
+  constructor (tripContainer, pointsModel) {
+    this.#tripContainer = tripContainer;
+    this.#pointsModel = pointsModel;
+  }
 
-    for (let i = 1; i < this.tripPoints.length; i++) {
-      render(new PointView(this.tripPoints[i]), this.tripList.getElement());
+  init = () => {
+    this.#tripPoints = [...this.#pointsModel.points];
+
+    this.#renderList();
+
+  };
+
+  #renderPoint = (point) => {
+    const pointComponent = new PointView(point);
+    const pointEditComponent = new PointEditView(point);
+
+    const replacePointToEditForm = () => {
+      this.#tripList.element.replaceChild(pointEditComponent.element, pointComponent.element);
+    };
+
+    const replaceEditFormToPoint = () => {
+      this.#tripList.element.replaceChild(pointComponent.element, pointEditComponent.element);
+    };
+
+    const onEscKeyDown = (evt) => {
+      if (evt.key === 'Escape' || evt.key === 'Esc') {
+        evt.preventDefault();
+        replaceEditFormToPoint();
+        document.removeEventListener('keydown', onEscKeyDown);
+      }
+    };
+
+    pointComponent.element.querySelector('.event__rollup-btn').addEventListener('click', () => {
+      replacePointToEditForm();
+      document.addEventListener('keydown', onEscKeyDown);
+    });
+
+    pointEditComponent.element.querySelector('form').addEventListener('submit', (evt) => {
+      evt.preventDefault();
+      replaceEditFormToPoint();
+      document.removeEventListener('keydown', onEscKeyDown);
+    });
+
+    pointEditComponent.element.querySelector('.event__rollup-btn').addEventListener('click', () => {
+      replaceEditFormToPoint();
+      document.removeEventListener('keydown', onEscKeyDown);
+    });
+
+    render(pointComponent, this.#tripList.element);
+  };
+
+  #renderList = () => {
+    render(new SortView(), this.#tripContainer);
+    render(this.#tripList, this.#tripContainer);
+
+    for (let i = 0; i < this.#tripPoints.length; i++) {
+      this.#renderPoint(this.#tripPoints[i]);
     }
```

Comment:
Здесь уже лучше заменить на forEach
```suggestion
    this.#tripPoints.forEach(this.#renderPoint);
```